src/jit/emitxarch.cpp

```diff
@@ -9929,7 +9929,7 @@ BYTE* emitter::emitOutputRI(BYTE* dst, instrDesc* id)
                 break;
 
             case IF_RRW_CNS:
-                assert(id->idGCref() == GCT_BYREF);
+                assert((id->idGCref() == GCT_BYREF) || (id->idGCref() == GCT_GCREF));
```

Comment:
I was hitting this with an earlier set of changes; the current changes no longer trigger the assert. I've restored it to its original version.